stl/inc/filesystem

```diff
@@ -1377,6 +1377,36 @@ namespace filesystem {
 #endif // ^^^ !_HAS_CXX20 ^^^
 
         _NODISCARD_FRIEND path operator/(const path& _Left, const path& _Right) { // append a pair of paths together
+            const auto _Right_size  = _Right._Text.size();
+            const auto _Right_first = _Right._Text.data();
+            const auto _Right_last  = _Right_first + _Right_size;
+
+            // Handle the most common case
+            if (!_Has_drive_letter_prefix(_Right_first, _Right_last) && !_Is_slash(*_Right_first)) {
+                // !has_root_name(_Right) && !has_root_directory(_Right)
+                const auto _Left_size  = _Left._Text.size();
+                const auto _Left_first = _Left._Text.data();
+                const auto _Left_last  = _Left._Text.data() + _Left_size;
+
+                // Appending a slash to "X:" would've made it an absolute path
+                const bool _Left_is_just_drive = _Left_size == 2 && _Is_drive_prefix(_Left_first);
+                const bool _Is_slash_needed    = _Left_size != 0 && !_Left_is_just_drive && !_Is_slash(_Left_last[-1]);
+
+                const auto _Total_size = _Left_size + _Is_slash_needed + _Right_size;
+
+                path _Tmp;
+                _Tmp._Text._Resize_and_overwrite(_Total_size, [&](wchar_t* _Ptr, size_t _Size) {

```

Comment:
It probably doesn't matter at all for optimized compiles, but everything we capture here (`_(Left|Right)_(first|size)` and `_Is_slash_needed`) is notably not-more-than-pointer-sized and unmodified. Maybe capturing copies will help alias analysis?
```suggestion
                _Tmp._Text._Resize_and_overwrite(_Total_size, [=](wchar_t* _Ptr, size_t _Size) {
```

Reply:
That's a reasonable motivation. The optimizer can't see through data members, but it does have some knowledge of when things are address-taken.